package/MDAnalysis/analysis/bat.py

```diff
@@ -0,0 +1,503 @@
+to be complete, non-redundant, and minimize correlations between degrees
+of freedom. Complete and non-redundant means that for N atoms there will
+be 3N Cartesian coordinates and 3N BAT coordinates. Correlations are
+minimized by using improper torsions described in [Hikiri2016]_.
+
+More specifically, bond refers to the bond length, or distance between
+a pair of bonded atoms. Angle refers to the bond angle, the angle between
+a pair of bonds to a central atom. Torsion refers to the torsion angle.
+For a set of four atoms a, b, c, and d, a torsion requires bonds between
+a and b, b and c, and c and d. The torsion is the angle between a plane
+containing atoms a, b, and c and another plane containing b, c, and d.
+For a set of torsions that share atoms b and c, one torsion is defined as
+the primary torsion. The others are defined as improper torsions, differences
+between the raw torsion angle and the primary torsion. This definition reduces
+the correlation between the torsion angles.
+
+Each molecule also has six external coordinates that define its translation and
+rotation in space. The three Cartesian coordinates of the first atom are the
+molecule's translational degrees of freedom. Rotational degrees of freedom are
+specified by the axis-angle convention. The rotation axis is a normalized vector
+pointing from the first to second atom. It is described by the polar angle,
+:math:`phi`, and azimuthal angle, :math:`theta`. :math:`omega` is a third angle
+that describes the rotation of the third atom about the axis.
+
+This module was adapted from AlGDock [Minh2020]_.
+
+
+See Also
+--------
+:class:`~MDAnalysis.analysis.dihedrals.Dihedral`
+   class to calculate dihedral angles for a given set of atoms or residues
+:func:`MDAnalysis.lib.distances.calc_dihedrals()`
+   function to calculate dihedral angles from atom positions
+
+
+Example applications
+--------------------
+
+The :class:`~MDAnalysis.analysis.bat.BAT` class defines bond-angle-torsion
+coordinates based on the topology of an atom group and interconverts between
+Cartesian and BAT coordinate systems. For example, we can determine internal
+coordinates for residues 5-10 of adenylate kinase (AdK). The trajectory is
+included within the test data files::
+
+   import MDAnalysis as mda
+   from MDAnalysisTests.datafiles import PSF, DCD
+   import numpy as np
+
+   u = mda.Universe(PSF, DCD)
+
+   # selection of atomgroups
+   selected_residues = u.select_atoms("resid 5-10")
+
+   from MDAnalysis.analysis.bat import BAT
+   R = BAT(selected_residues)
+
+   # Calculate BAT coordinates for a trajectory
+   R.run()
+
+   # Reconstruct Cartesian coordinates from BAT coordinates
+   # of the first frame
+   bat = R.bat[0]
+   XYZ = R.Cartesian(bat)
+
+   # The difference between the original and reconstructed coordinates
+   # should be zero.
+   print(np.sum(np.abs(XYZ - selected_residues.positions)>1E-6))
+
+After R.run(), the coordinates can be accessed with :attr:`R.bat`.
+
```

Comment:
The description of the `bat` data structure can come under the actual attribute docs or as a separate subsection in the module docs. 

1. You can  add it under the attribute documentation (see my comment below on _Document the attributes that contain data_). Then whenever you link to ``attr:bat``, a user will immediately know what's in there. 

2. If you decide to make it a subsection in the module docs, add a sub header, put a
    ```reST
    .. _bat-datastructure-description:
    
    BAT data structure description
    ------------------------------
    
    The :attr:`BAT.bat` attribute contains the bond-angle-torsion coordinates for all frames
    in the trajectory. It is structures as NumPy :class:`numpy.ndarray` with shape `(n_frames, 3N)`
    where ...
    ```

    Then in the attribute text add a ``:ref:...`` link to the section into the attribute docs like so
    ```reST

    .. autoclass:: BAT 
        :members: 
        :inherited-members: 
  
        .. attribute:: bat 
      
         Bond-angle-torsions internal coordinates for all trajectory frames as a numpy array
         with shape (`n_frames, 3*N)` for the `N` atoms in the :class:`AtomGroup` `ag`.
         See :ref:`bat-datastructure-description` for details of the data structure.
    ```

Reply:
Okay, I took option 1 based on your other comment on "Document the attributes that contain data"